image-download

```diff
@@ -68,20 +73,21 @@ def find(name, stores, latest, quiet):
             ai = socket.getaddrinfo(url.hostname, url.port or defport, socket.AF_INET, 0, socket.IPPROTO_TCP)
         except socket.gaierror:
             ai = []
-            message = store
 
         for (family, socktype, proto, canonname, sockaddr) in ai:
-            message = "{scheme}://{0}:{1}{path}".format(*sockaddr, scheme=url.scheme, path=url.path)
```

Comment:
So the reason we were showing the IP address is because we clearly anticipated multiple IPs per name here.  If that's no longer the case, then we ought to simplify this code much more substantially.

Reply:
I'm not sure about the history here. I think that was from a time when we had OpenShift running ourselves, and used http://cockpit-images.verify.svc.cluster.local . This was dropped in e6d417cc65644a8195c0c8130840a75d48d20374 . But I honestly don't know why it was necessary in the first place -- services and routes should take care of the load balancing, and I'm not sure why one needs to pin down an IP for that. I'll try to drop it.

Reply:
Ah, of course -- we use that with `curl --resolve` (so that we can use the "cockpit-test" SNI), which requires an IP. So this has to do a DNS resolution.